typescript error `Property 'theme' is missing`

### Current behaviour
As reported and explained here: https://github.com/microsoft/rnx-kit/issues/2247#issuecomment-1450186908
```
src/App.native.tsx:196:12 - error TS2741: Property 'theme' is missing in type '{ icon: string; size: number; onPress: () => void; }' but required in type 'Pick<Props, "testID" | "disabled" | "hitSlop" | "onLayout" | "pointerEvents" | "removeClippedSubviews" | "style" | "nativeID" | "collapsable" | "needsOffscreenAlphaCompositing" | ... 72 more ... | "size">'.

196           <IconButton
               ~~~~~~~~~~

  ../../node_modules/react-native-paper/lib/typescript/components/TouchableRipple/TouchableRipple.native.d.ts:13:5
    13     theme: InternalTheme;
           ~~~~~
    'theme' is declared here.


Found 1 error in src/App.native.tsx:196
```

### Expected behaviour
No typescript errors

### How to reproduce?
Add ` "moduleSuffixes": [".android", ".native", ""]` to tsconfig

### What have you tried so far?
replace content of `TouchableRipple.native.tsx`
```ts
type Props = React.ComponentProps<typeof Pressable> & {
  ...
  children: React.ReactNode;
  style?: StyleProp<ViewStyle>;
  theme: InternalTheme;
};
```

by `TouchableRipple.tsx` content, where theme is optional
```ts
export type Props = React.ComponentPropsWithRef<typeof Pressable> & {
  ...
  children: React.ReactNode;
  style?: StyleProp<ViewStyle>;
  theme?: ThemeProp;
};
```

### Your Environment

| software                      | version
| --------------------- | -------
| react-native             | 0.71.3
| react-native-paper  | 5.3.1
| node                        | 16.17.0
| expo sdk                  | 48.0.0
| typescript                | 4.9.4

--- a/src/components/TouchableRipple/TouchableRipple.native.tsx
+++ b/src/components/TouchableRipple/TouchableRipple.native.tsx
@@ -10,22 +10,24 @@ import {
 } from 'react-native';
 
 import { useInternalTheme } from '../../core/theming';
-import type { InternalTheme } from '../../types';
+import type { ThemeProp } from '../../types';
 import { getTouchableRippleColors } from './utils';
 
 const ANDROID_VERSION_LOLLIPOP = 21;
 const ANDROID_VERSION_PIE = 28;
 
-type Props = React.ComponentProps<typeof Pressable> & {
+export type Props = React.ComponentProps<typeof Pressable> & {
   borderless?: boolean;
   background?: PressableAndroidRippleConfig;
+  centered?: boolean;
   disabled?: boolean;
   onPress?: (e: GestureResponderEvent) => void | null;
+  onLongPress?: (e: GestureResponderEvent) => void;
   rippleColor?: string;
   underlayColor?: string;
   children: React.ReactNode;
   style?: StyleProp<ViewStyle>;
-  theme: InternalTheme;
+  theme?: ThemeProp;
 };
 
 const TouchableRipple = ({

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -16,6 +16,7 @@
       "esnext",
       "dom"
     ],
+    "moduleSuffixes": [".ios", ".native", ".android", ""],
     "module": "esnext",
     "moduleResolution": "node",
     "noFallthroughCasesInSwitch": true,

